javalib/src/main/scala/java/nio/file/Files.scala

```diff
@@ -81,8 +82,15 @@ object Files {
         throw new FileAlreadyExistsException(targetFile.getAbsolutePath())
       }
 
-    try copy(in, out)
-    finally out.close()
+    try {
+      val copyResult = copy(in, out)
+      // Make sure that created file has correct permissions
```

Comment:
On MacOS M1 the files created by copying were given empty permissions, this should match the permission of files created by the JVM